powerline_prompt.lua

```diff
@@ -43,16 +42,33 @@ local segment = {
 -- Sets the properties of the Segment object, and prepares for a segment to be added
 ---
 local function init()
-	cwd = clink.get_cwd()
-	if powerline_config_prompt_type == promptTypeFolder then
+    -- fullpath
+    cwd = clink.get_cwd()
+
+    -- show just current folder
+    if powerline_config_prompt_type == promptTypeFolder then
 		cwd =  get_folder_name(cwd)
-	else 
-		if powerline_config_prompt_useHomeSymbol then 
-			if string.find(cwd, clink.get_env("HOME")) then 
-				cwd = string.gsub(cwd, clink.get_env("HOME"), homeSymbol)
-			end
-		end
-	end
+    else    
+        local git_dir = get_git_dir()
```

Comment:
that is a piccle, as the git component is quite crucial maybe we move it to very beginning to override it so that no addon gonna have same issue. No its not as clean at it could be, but would resolve it and ensure it wont randomly break?